src/assisted_installer_controller/assisted_installer_controller.go

```diff
@@ -70,22 +75,41 @@ func NewController(log *logrus.Logger, cfg ControllerConfig, ops ops.Ops, ic inv
 	}
 }
 
-func (c *controller) WaitAndUpdateNodesStatus() {
+func (status *ControllerStatus) Error() {
+	atomic.AddUint32(&status.errCounter, 1)
+}
+
+func (status *ControllerStatus) HasError() bool {
+	return atomic.LoadUint32(&status.errCounter) > 0
+}
+
+func (c *controller) WaitAndUpdateNodesStatus(status *ControllerStatus) {
 	c.log.Infof("Waiting till all nodes will join and update status to assisted installer")
-	ignoreStatuses := []string{models.HostStatusDisabled,
-		models.HostStatusError, models.HostStatusInstalled}
+	ignoreStatuses := []string{models.HostStatusDisabled, models.HostStatusInstalled}
+	var hostsInError int
 	for {
 		time.Sleep(GeneralWaitInterval)
 		ctx := utils.GenerateRequestContext()
 		log := utils.RequestIDLogger(ctx, c.log)
+
 		assistedInstallerNodesMap, err := c.ic.GetHosts(ctx, log, ignoreStatuses)
 		if err != nil {
 			log.WithError(err).Error("Failed to get node map from inventory")
 			continue
 		}
+		errNodesMap := c.ic.FilterHostsByStatus(assistedInstallerNodesMap, models.HostStatusError)
```

Comment:
I think will be very helpful to be able to filter by status as a list. What do u think?

Reply:
I don't know because we haven't got a real use case for it, but it can be done. I'll fix